acceptance/suites/tests/030-tk-auth/010-tk-auth-smoke.rb

```diff
@@ -0,0 +1,146 @@
+confine :except, :platform => 'windows'
+
+test_name "TK Auth Smoke Test Presuite" do
+  
+  #TODO: skip if pe
+  step 'Configure the server setting' do
+    # so that we don't have to say puppet agent -t --server={fqdn}
+    master_fqdn=on(master, 'facter fqdn').stdout.chomp
+    hosts.each do |h|
+      on(h, "puppet config set server #{master_fqdn}") 
+    end
+  end
+
+  step 'Install required modules available on puppet forge' do
+    on(master, 'puppet resource package git ensure=present')
+    on(master, 'puppet module install puppetlabs-vcsrepo')
+    on(master, 'puppet module install puppetlabs-stdlib')
+    on(master, 'puppet module install puppetlabs-concat')
+    on(master, 'puppet module install puppetlabs-hocon')
+  end
+
+  #TODO: skip if present 
+  step 'Acquire a git auth key ' do
+    gitsshkey=File.file?("#{ENV['HOME']}/.ssh/id_rsa") ? "#{ENV['HOME']}/.ssh/id_rsa" : File.expand_path('~/.ssh/id_rsa-jenkins')
+    master.do_scp_to(gitsshkey, '/root/.ssh/gitkey',options)
+    on(master, 'eval $(ssh-agent)')
+  end
+
+  #TODO: skip if present
+  step 'Install hocon gem' do
+    #TODO: Is there a way we can calculate this location from puppet config?
+    on(master, '/opt/puppetlabs/puppet/bin/gem install hocon')
+  end
+
+#TODO: Make this look in forge first, and then proceed to git if forge fails.
+  step 'Install puppetlabs-hocon and puppetlabs-http_authorization modules from git' do
+    environmentpath=on(master, 'puppet config print environmentpath').stdout.chomp
+    environment='production'
+    $manifest_path=[environmentpath,environment,'manifests','site.pp'].join('/')
+    modulepath=[environmentpath,environment,'modules'].join('/')
+    manifest_text = <<-END
+#vcsrepo { '#{modulepath}/hocon' :
+#  ensure    => present,
+#  provider  => git,
+#  source    => 'git@github.com:puppetlabs/puppetlabs-hocon.git',
+#  user      => 'root',
+#  revision  => 'master',
+#  identity  => '/root/.ssh/gitkey',
+#  }
+
+vcsrepo { '#{modulepath}/http_authorization' :
+  ensure    => present,
+  provider  => git,
+  source    => 'git@github.com:puppetlabs/puppetlabs-http_authorization',
+  user      => 'root',
+  revision  => 'master',
+  identity  => '/root/.ssh/gitkey',
+  }
+  
+    END
+    create_remote_file(master, $manifest_path, manifest_text)
+    on(master, "chmod +r #{$manifest_path}")
+    on(master, 'puppet agent -t', :acceptable_exit_codes => [0,2])
+  end
+
+  #TODO: file a ticket?
+  #check if version is 9.4 and fail if version isn't so we are forced to fix this 
+  #when it is fixed upstream.
+  step 'WORKAROUND: prerelease puppetlabs-hocon...' do
+    on(master, 'perl -pi -e \'s/"version": "0.9.2"/"version": "0.9.4"/g\' /etc/puppetlabs/code/environments/production/modules/hocon/metadata.json')
```

Comment:
Is this because `puppetlabs-http_authorization` currently depends upon a version of `puppetlabs-hocon` which doesn't exist yet?